src/emulator/gui/include/gui/imgui_impl.h

```diff
@@ -0,0 +1,36 @@
+// Vita3K emulator project
+// Copyright (C) 2018 Vita3K team
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+#pragma once
+
+#include <host/state.h>
+
+#include <glutil/gl.h>
+
+// clang-format off
```

Comment:
I've seen this in a few places. I'm not saying it's wrong, but why is it needed?

Side note: My preference is to put our headers before 3rd party ones (separated by a newline) as this should encourage ours to be self-contained and not rely on other headers coming first.

Reply:
It's needed because imgui.h need to be included before any other imgui-related files are included (else you get a compile error) and clang-format changes their order without this.

Reply:
And I definitely agree with the side note, I'll change their order.